### PR TITLE
Fix pre-release candidate publishing

### LINKED ISSUE #1135: Fix pre-release candidate publishing
# 🐛 Summary #

The publish public package workflow no longer supports publishing pre-release candidates.

## To reproduce ##

1. Look at workflow code
1. Note where James commented this code out during previous bug-fixing work but never uncommented it when done
2. Say "oops" out loud

## Expected behavior ##

This workflow should publish release and pre-release.

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

Fixes the publish to PSGallery workflow so that it (once again) can publish a release candidate.

## 💭 Motivation and context ##

Closes #1135 

## 🧪 Testing ##

Tested publishing to PSGallery.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords]- [ ] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [x] PR passed smoke test check.
- [x] Feature branch has been rebased against changes from parent branch, as needed
- [x] Resolved all merge conflicts on branch
- [x] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: .github/workflows/publish_public_package.yaml [hunk 1]
```diff
@@ -83,22 +84,26 @@ jobs:
...
-          #   $Parameters.Add('PrereleaseTag', '${{ inputs.PrereleaseTag }}')
-          # }
-          # if (-Not [string]::IsNullOrEmpty('${{ inputs.OverrideModuleVersion }}'))
-          # {
-          #   $Parameters.Add('OverrideModuleVersion', '${{ inputs.OverrideModuleVersion }}')
-          # }
+          if ('true' -eq '${{ inputs.IsPrerelease }}')
+          {
+            Write-Host "Adding IsPrerelease"
+            Write-Host ${{ inputs.IsPrerelease }}
```
Comment (adhilto) [inline .github/workflows/publish_public_package.yaml:90]: I'm pretty sure our PowerShell linter enforces this, but as this PowerShell inside a YAML file, the linter didn't check it.

Per [AvoidUsingWriteHost](https://learn.microsoft.com/en-us/powershell/utility-modules/psscriptanalyzer/rules/avoidusingwritehost?view=ps-modules), "The use of Write-Host is greatly discouraged." It looks like these are basically just logging message, in which case `Write-Verbose` is what Microsoft would recommend we use.

File: .github/workflows/publish_public_package.yaml [hunk 2]
```diff
@@ -83,22 +84,26 @@ jobs:
...
+            $Parameters.Add('PrereleaseTag', '${{ inputs.PrereleaseTag }}')
+          }
+          if (-Not [string]::IsNullOrEmpty('${{ inputs.OverrideModuleVersion }}'))
+          {
+            Write-Host "Adding OverrideModuleVersion"
+            Write-Host ${{ inputs.OverrideModuleVersion }}
```
Comment (adhilto) [inline .github/workflows/publish_public_package.yaml:96]: More `Write-Host` calls.

**Review by adhilto — COMMENTED** [review]
Not sure if/how I can test this, but I did identify a few style things by reviewing the code changes. Note that there are other style issues in this workflow, but I only commented on the style in the code changed in this PR.

**Review by dagarwal-mitre — APPROVED** [review]

File: .github/workflows/publish_public_package.yaml (hunk 1 above)
Comment (james-garriss) [reply .github/workflows/publish_public_package.yaml]: You are correct, Write-Host is discouraged.  Unfortunately, there are no alternatives that I am aware of for GitHub Action worksflows.  I've tested this all before, but I just tested it again.  Comments like this:

```
Write-Verbose "WV: Adding IsPrerelease"
Write-Verbose ${{ inputs.IsPrerelease }}
```

do not actually publish anything to the workflow console logs.  And a comment like this:

```
Write-Verbose "WV: Checking for prerelease with required version: " + $Version
```

actually throws an error and breaks the workflow. 

See here for example: https://github.com/cisagov/ScubaGear/actions/runs/9454334557/job/26041655409

 If you know of another `Write-<something>` that actually works, I would be happy to switch, but AFAICT there are no options.

File: .github/workflows/publish_public_package.yaml (hunk 2 above)
Comment (james-garriss) [reply .github/workflows/publish_public_package.yaml:98]: See above.

File: .github/workflows/publish_public_package.yaml (hunk 1 above)
Comment (james-garriss) [reply .github/workflows/publish_public_package.yaml]: Ran a test on the various Write-*'s and found that host, output, and warning work.  Error throws an error.  Switch all hoss to output.  Also fixed a concatenation typo.

**Comment by james-garriss** [comment]
@nanda-katikaneni , this one is ready.

**Pull request merged**